lib/run-datadog.sh

```diff
@@ -6,87 +6,78 @@ DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/datadog}"
 
 start_datadog() {
   pushd $DATADOG_DIR
-    export DD_LOG_FILE=$DATADOG_DIR/dogstatsd.log
-    export DD_API_KEY
-    export DD_DD_URL=${DD_DD_URL:-https://app.datadoghq.com}
-    export DD_ENABLE_CHECKS="${DD_ENABLE_CHECKS:-true}"
-    export DOCKER_DD_AGENT=yes
-    export LOGS_CONFIG_DIR=$DATADOG_DIR/dist/conf.d/logs.d
-    export LOGS_CONFIG
+    export HOME=${HOME:-/home/vcap}
 
-    # create and configure set /conf.d if integrations are enabled
-    if [ "$DD_ENABLE_CHECKS" = "true" ] || [ -n "$LOGS_CONFIG" ] ; then
-      mkdir $DATADOG_DIR/dist/conf.d
-      sed -i "s~# confd_path:.*~confd_path: $DATADOG_DIR/dist/conf.d~" $DATADOG_DIR/dist/datadog.yaml
-    fi
 
-    # add checks configs
-    if [ "$DD_ENABLE_CHECKS" = "true" ]; then
-      mv $DATADOG_DIR/conf.d/* $DATADOG_DIR/dist/conf.d
-    fi
+    echo "Fixing pkg config settings..."
+    pushd $DATADOG_DIR/opt/datadog-agent/embedded/lib/pkgconfig/
+      OLD_PREFIX="prefix=/opt/datadog-agent/embedded"
+      NEW_PREFIX="prefix=$DATADOG_DIR/opt/datadog-agent/embedded"
+      for file in $(ls); do
+        if [ -f "${file}" -a ! -L "${file}" ]; then
+          sed -i "s~${OLD_PREFIX}~${NEW_PREFIX}~" ${file}
+        fi
+      done
+    popd
+
+    echo "Fixing python prefixes..."
+    pushd $DATADOG_DIR/opt/datadog-agent/embedded/lib/python2.7/
+      echo "import site" >> sitecustomize.py
+      echo "site.addsitedir('$DATADOG_DIR/opt/datadog-agent/embedded/lib/python2.7/site-packages')" >> sitecustomize.py
+      echo "site.PREFIXES.append('$DATADOG_DIR/opt/datadog-agent/embedded/')" >> sitecustomize.py
+    popd
+
+    # Setup LD_LIBRARY_PATH
+    LD_LIBRARY_PATH="$DATADOG_DIR/opt/datadog-agent/embedded/lib:$LD_LIBRARY_PATH"
+    for package_lib_dir in $(ls -d $DATADOG_DIR/opt/datadog-agent/embedded/lib 2>/dev/null); do
+        LD_LIBRARY_PATH="$package_lib_dir:$LD_LIBRARY_PATH"
+    done
+    for package_lib_dir in $(ls -d $DATADOG_DIR/opt/datadog-agent/embedded/lib/python*/lib-dynload 2>/dev/null); do
+        LD_LIBRARY_PATH="$package_lib_dir:$LD_LIBRARY_PATH"
+    done
+    for package_lib_dir in $(ls -d $DATADOG_DIR/opt/datadog-agent/embedded/lib/python*/site-packages 2>/dev/null); do
+        LD_LIBRARY_PATH="$package_lib_dir:$LD_LIBRARY_PATH"
+    done
+
+    # configure confd and checksd
+    export DD_CONFD_PATH=${DD_CONFD_PATH:-"$DATADOG_DIR/etc/datadog-agent/conf.d"}
+  	export DD_ADDITIONAL_CHECKSD=${DD_ADDITIONAL_CHECKSD:-"$DATADOG_DIR/etc/datadog-agent/checks.d"}
+
+    # Setup PYTHONPATH and PYTHONHOME
```

Comment:
Let's create a method for all this py2 config logic so that we can easily switch to p3 later too

Reply:
I don't know exactly yet what switching to py3 will entail so I don't think there is value in encapsulating that in a function.

Reply:
ok